ledger-api/grpc-definitions/com/daml/ledger/api/v1/experimental_features.proto

```diff
@@ -67,3 +68,8 @@ enum CommandDeduplicationType {
   // Duplicate commands are always reported synchronously via a synchronous gRPC error on the command submission.
   SYNC_ONLY = 2;
 }
+
+message ContractIdFeatures {
+  bool v0_supported = 1;
+  bool v1_non_suffixed_supported = 2;
+}
```

Comment:
@SamirTalwar-DA please add comments here that are detailed enough for any of our engineers to undersand what the two features are and which ones are expected to be supported.

Reply:
I'd expect something along one or two sentences per field and a pointer to the details which are likely present in either the test code or the LF infrastructure code.

Reply:
Also: right now the default value of this message is such that none of the two modes are supported, which means that by default none of the corresponding tests are run. Is there the option to choose a reasonable default that works for Canton out of the box, and thereby ensure that on upgrade the test tool will always test their supported contact-ids?

Reply:
fyi v0 contract IDs  will be dropped with Daml 2.0, so you could drop the corresponding tests. 
language-team can take care of it, when we effectively drop the support.

Reply:
Probably worth add a referent to the [Contract ID spec](daml-lf/spec/contract-id.rst).

Reply:
Canton and KV support distinct contract IDs (non_suffixed vs suffixed), so it seems difficult to have a reasonable default.

Reply:
Thanks for the input @remyhaemmerle-da . The `non_suffixed vs suffixed` probably means that there should be another field or an enum that allows to represent Canton's config as well, shouldn't there?

Reply:
@meiersi-da Good point.

We need: 
-  a features for `sandbox-on-x`, that check that both kinds of CIDs are accepted. 
 - a feature for canton, that checks that suffixed CID are accepted and non-suffixed ones are rejected.

Reply:
KV also supports both kinds of v1 CIDs. I am not sure if that's intentional, but it does.

Reply:
(You can see this in [the _ledger-on-memory_ diff](https://github.com/digital-asset/daml/pull/12454/files#diff-c3adf0e08faaddf3fdc7b9efd7bcd0fcb9bca1b5bbafe0c4c33c369fd); we used to explicitly enable the tests accordingly.)

```bazel
        "--additional=ContractIdIT:RejectV0,ContractIdIT:AcceptSuffixedV1,ContractIdIT:AcceptNonSuffixedV1",
```

Reply:
@remyhaemmerle-da, @meiersi-da, please take another look when you have a moment. I think it's clearer now.

I removed the v0 contract ID feature flag and the associated test cases.

Reply:
This doesn’t make sense to me. We should still test that v0 contract ids are rejected especially if we’re dropping support for them. What we can drop is the test that they’re accepted.